From 0b98b920c829cb19f52adc699f25f370a559cf68 Mon Sep 17 00:00:00 2001
From: Vlad Yasevich <vyasevic@redhat.com>
Date: Fri, 29 Aug 2014 19:01:15 -0400
Subject: [net] bridge: Turn flag change macro into a function

Message-id: <1409338885-30414-4-git-send-email-vyasevic@redhat.com>
Patchwork-id: 89905
O-Subject: [RHEL7 PATCH 03/13] bridge: Turn flag change macro into a function.
Bugzilla: 896669
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>

From: Vlad Yasevich <vyasevic@redhat.com>

Turn the flag change macro into a function to allow
easier updates and to reduce space.

Acked-by: Michael S. Tsirkin <mst@redhat.com>
Signed-off-by: Vlad Yasevich <vyasevic@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 63c3a622dd020dd16959fdb1bfa79710dd702420)
Signed-off-by: Vladislav Yasevich <vyasevic@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/bridge/br_sysfs_if.c b/net/bridge/br_sysfs_if.c
index 2a2cdb7..265a878 100644
--- a/net/bridge/br_sysfs_if.c
+++ b/net/bridge/br_sysfs_if.c
@@ -41,20 +41,27 @@ static ssize_t show_##_name(struct net_bridge_port *p, char *buf) \
 }        \
 static int store_##_name(struct net_bridge_port *p, unsigned long v) \
 {        \
- unsigned long flags = p->flags;    \
- if (v)       \
-  flags |= _mask;     \
- else       \
-  flags &= ~_mask;    \
- if (flags != p->flags) {    \
-  p->flags = flags;    \
-  br_ifinfo_notify(RTM_NEWLINK, p);  \
- }       \
- return 0;      \
+ return store_flag(p, v, _mask);    \
 }        \
 static BRPORT_ATTR(_name, S_IRUGO | S_IWUSR,   \
      show_##_name, store_##_name)
 
+static int store_flag(struct net_bridge_port *p, unsigned long v,
+        unsigned long mask)
+{
+ unsigned long flags = p->flags;
+
+ if (v)
+  flags |= mask;
+ else
+  flags &= ~mask;
+
+ if (flags != p->flags) {
+  p->flags = flags;
+  br_ifinfo_notify(RTM_NEWLINK, p);
+ }
+ return 0;
+}
 
 static ssize_t show_path_cost(struct net_bridge_port *p, char *buf)
 {
-- 
1.7.1